cppcache/src/ThinClientPoolDM.cpp

```diff
@@ -1428,10 +1428,12 @@ GfErrType ThinClientPoolDM::sendSyncRequest(
           }
           excludeServers.insert(ServerLocation(ep->name()));
           if (error == GF_IOERR) {
-            auto sl = std::make_shared<BucketServerLocation>(ep->name());
-            LOGINFO("Removing bucketServerLocation %s due to GF_IOERR",
-                    sl->toString().c_str());
-            m_clientMetadataService->removeBucketServerLocation(sl);
+            if (m_clientMetadataService != nullptr) {
```

Comment:
The one style thing I would point out is that `std::unique_ptr` explicitly converts to `true` if set or `false` if is `nullptr` specifically for the usage in if statements like this.
```c++
if (m_clientMetadataService) {
  m_clientMetadataService->doSomething();
}
```

Reply:
ok